beacon_node/beacon_chain/src/validator_custody.rs

```diff
@@ -0,0 +1,433 @@
+use std::{
+    collections::{BTreeMap, HashMap},
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use parking_lot::RwLock;
+
+use ssz_derive::{Decode, Encode};
+use tokio::sync::broadcast::{channel, Receiver, Sender};
+use types::{ChainSpec, Epoch, EthSpec, Slot};
+
+const CHANNEL_CAPACITY: usize = 10;
+
+/// TODO(pawan): this currently just registers increases in validator count.
+/// Does not handle decreasing validator counts
+#[derive(Default, Debug)]
+struct ValidatorRegistrations {
+    /// Set of all validators that is registered to this node along with its effective balance
+    /// in increments of `BALANCE_PER_ADDITIONAL_CUSTODY_GROUP`
+    ///
+    /// Key is validator index and value is effective_balance // BALANCE_PER_ADDITIONAL_CUSTODY_GROUP.
```

Comment:
Any reason why we don't keep the raw effective balance here? Don't we potentially end up with lower custody requirements if we do floor division first? 

e.g. Say if i have 2 vals with 63 eth each, then 
- divide individually then sum: 2 custody units
- summing up first then divide: 3 custody unit

Reply:
ah sorry just read @dknopik 's comment [above](https://github.com/sigp/lighthouse/pull/7578#discussion_r2133988876)! nvm